### PR TITLE
Feature/gis 288

### PR DESCRIPTION
**Pull request by GlennDigipolis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -313,6 +316,12 @@ class ExampleComponent {
     @Input() sortBy = '';
     /* Use geolocation when the component finished loading */
     @Input() locateUserOnInit = false;
+    /* Set time to wait after user stops typing before triggering a search */
+    @Input() debounceTime = 50;
+    /* whether or not to return a single cascading result */
+    @Input() cascadingReturnSingle = true;
+    /* Cascading configuration for doing reverse lookups by coordinates */
+    @Input() cascadingRules: Array<CascadingRulesModel> = this.locationPickerHelper.getDefaultCascadingConfig();
```
Comment (jsebrech) [inline README.md:324]: instead of naming these "cascading..." might they be better named "coordinates..."? It would be a lot more intuitive for the user. Also, why isn't "cascadingLimit" documented?

File: projects/example/src/app/app.component.ts [hunk 2]
```diff
@@ -10,16 +9,8 @@ export class AppComponent implements OnInit {
...
-    layer: {
-      name: 'Satellite View',
-      url: 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}'
-    },
-    buttonLabel: 'Luchtfoto'
-  };
-
   ngOnInit(): void {
-    this.baseUrl = 'https://locationpicker-app1-o.antwerpen.be';
+    this.baseUrl = 'http://localhost:3000';
```
Comment (jsebrech) [inline projects/example/src/app/app.component.ts:13]: is this change intentional? might be handier to point to the server instance

File: projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts [hunk 3]
```diff
@@ -22,130 +23,139 @@ import {CoordinateModel} from '../types/coordinate.model';
...
+   * @param layers (the layers to look for locations in, only used for querying searchLocations)
+   * @param prioritizelayer (the layer to boost)
+   * @param sort (key to sort results by)
+   * @param cascadingReturnSingle (whether or not to return a single cascading result)
+   * @param cascadingRules (the configuration for doing reverse lookups by coordinates)
+   *
+   * @return Observable<LocationModel[] | AddressModel[] | CoordinateModel[]>
+   */
+  delegateSearch(
+    search: string,
```
Comment (jsebrech) [inline projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts:55]: at some point it would be better to refactor this into a config object instead of tacking on an endless list of parameters

File: projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts [hunk 4]
```diff
@@ -52,39 +53,32 @@ export class NgxLocationPickerService {
...
       const locationQuery: LocationQueryModel = {
-        layers,
-        limit,
-        search,
-        prioritizelayer,
-        sort
+        layers: search.layers,
+        limit: search.limit,
+        search: search.search,
+        prioritizelayer: search.prioritizelayer,
```
Comment (jsebrech) [inline projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts:80]: what happened to the default value of prioritizeLayer?

File: projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts [hunk 5]
```diff
@@ -22,130 +24,132 @@ import {CoordinateModel} from '../types/coordinate.model';
...
+   * @param sort (key to sort results by)
+   * @param cascadingReturnSingle (whether or not to return a single cascading result)
+   * @param cascadingRules (the configuration for doing reverse lookups by coordinates)
+   *
+   * @return Observable<LocationModel[] | AddressModel[] | CoordinateModel[]>
+   */
+  delegateSearch(
+    search: DelegateSearchModel
+  ): Observable<LocationModel[] | AddressModel[] | CoordinateModel[]> {
+    search.prioritizelayer = search.prioritizelayer ? search.prioritizelayer : ['straatnaam'];
```
Comment (GlennDigipolis) [inline projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts:58]: Due to the refactor of the params to searchmodel added null check on prioritizelayer to determine if it has a value otherwise the default value is filled in.

I've noticed I still have to change the params documentation

File: README.md (hunk 1 above)
Comment (GlennDigipolis) [reply README.md:326]: Maybe a hybrid of both? cascadingCoordinateRules? Seems like the best name.

Comment (jsebrech) [reply README.md:326]: fine for me
